pallets/runtime/tests/src/staking/mod.rs

```diff
@@ -5819,3 +5819,33 @@ fn test_bond_too_small() {
         assert_ok!(both(90, 91, MinimumBond::get() + 1));
     });
 }
+
+#[test]
+fn validator_unbonding() {
+    ExtBuilder::default()
+        .validator_count(8)
+        .minimum_validator_count(1)
+        .build()
+        .execute_with(|| {
+            // Add a new validator successfully.
+            bond_validator_with_intended_count(50, 51, 500000, Some(2));
+            let entity_id = Identity::get_identity(&50).unwrap();
+            assert_permissioned_identity_prefs!(entity_id, 2, 1);
+            // Set minimum bond threshold to 50k POLYX
+            assert_ok!(Staking::set_min_bond_threshold(Origin::root(), 50000));
+            // Check that an error is given when unbonding beyond the minimum bond threshold as a validator
+            assert_noop!(
+                Staking::unbond(Origin::signed(51), 480000),
+                Error::<Test>::InvalidValidatorUnbondAmount
+            );
+            // Check that validator can unbond once it doesn't go below the minimum bond threshold
+            assert_ok!(Staking::unbond(Origin::signed(51), 80000));
+            assert_ok!(Staking::unbond(Origin::signed(51), 80000));
+            assert_ok!(Staking::unbond(Origin::signed(51), 80000));
+            assert_ok!(Staking::unbond(Origin::signed(51), 80000));
+            // Chill validator
+            assert_ok!(Staking::chill(Origin::signed(51)));
+            // After chilling validator checks that entity can unbond successfully
+            assert_ok!(Staking::unbond(Origin::signed(51), 160000));
```

Comment:
This test here should try unbonding 100% of the remaining "active" bonded amount and make sure active bonded amount is zero after this last unbond.